scripts/qiita-auto-processing

```diff
@@ -0,0 +1,271 @@
+        parameters = []
+        for dps in cmd.default_parameter_sets:
+            if dps.name in step['parameters_names']:
+                parameters.append({'id': dps.id, 'values': dps.values})
+
+        commands.append({
+            'command': cmd,
+            'command-name': cmd.name,
+            'previous-step': step['previous-step'],
+            'parent_artifact_name': step['parent_artifact_name'],
+            'input_name': step['input_name'],
+            'parameters': parameters})
+
+    # Step 3. Find all preparations/artifacts that we can add the pipeline
+    artifacs_all = [a for study in Study.iter()
+                    # loop over all artifacts of artifact_type with in study
+                    for a in study.artifacts(
+                        artifact_type=pipeline['artifact_type'])
+                    if _check_previous_command(
+                        pipeline['previous-step'], a.processing_parameters)]
+
+    # Step 4. Limit all_artifacts to those within restrictions
+    artifacts_compliant = []
+    for a in artifacs_all:
+        st = a.study.sample_template
+        pt = a.prep_templates[0]
+
+        # {'sandbox', 'awaiting_approval', 'private', 'public'}
+        if a.visibility in ('sandbox', 'awaiting_approval'):
+            continue
+
+        if pt.data_type() not in pipeline['data_type']:
+            continue
+
+        requirements = pipeline['requirements']
+        if 'sample' in requirements:
+            satisfied = True
+            for req in requirements['prep']:
+                if satisfied:
+                    if req['column'] not in st.categories():
+                        if req['equal']:
+                            satisfied = False
+                        continue
+                    sample_value = list(map(str.lower, set(
+                        st.get_category(req['column']).values())))
+                    if req['equal'] and sample_value != req['value']:
+                        satisfied = False
+                        continue
+                    elif not req['equal'] and sample_value == req['value']:
+                        satisfied = False
+                        continue
+            if not satisfied:
+                continue
+
+        if 'prep' in requirements:
+            satisfied = True
+            for req in requirements['prep']:
+                if satisfied:
+                    if req['column'] not in pt.categories():
+                        if req['equal']:
+                            satisfied = False
+                        continue
+                    prep_value = list(map(str.lower, set(
+                        pt.get_category(req['column']).values())))
+                    if req['equal'] and prep_value != req['value']:
+                        satisfied = False
+                        continue
+                    elif not req['equal'] and prep_value == req['value']:
+                        satisfied = False
+                        continue
```

Comment:
Any chance the code here could be put in a function that gets called for both cases?